staging/src/k8s.io/apiserver/pkg/cel/library/network_addrs.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package library
+
+import (
+	"net/netip"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/common/types/ref"
+
+	apiservercel "k8s.io/apiserver/pkg/cel"
+)
+
+// NetworkAddr provides a CEL function library extension of network address parsing functions.
```

Comment:
nit: It might just be me.. But I didn't immediately realize this type was for CIDR blocks. Should we somehow highlight in this description?

Reply:
Yep, we can either update the docs or change the name of this. Will see if I can find someone with better networking experience than I have to talk through different naming suggestions and come back to this